query/src/datasources/table/fuse/read.rs

```diff
@@ -66,10 +69,25 @@ impl FuseTable {
         let da = io_ctx.get_data_accessor()?;
         let arrow_schema = self.table_info.schema().to_arrow();
 
-        let stream = futures::stream::iter(iter);
-        let stream = stream.then(move |part| {
-            io::do_read(part, da.clone(), projection.clone(), arrow_schema.clone())
-        });
+        let mut iter = futures::stream::iter(iter);
+        let stream = stream! {
+            while let Some(part) = iter.next().await {
+                let mut source = ParquetSource::new(
+                    da.clone(),
+                    part.name.clone(),
+                    Arc::new(DataSchema::from(arrow_schema.clone())),
```

Comment:
may be we could avoid some `clone`s here, by moving `Arc::new(DataSchema::from(arrow_schema.clone()))` outside, and only clone the `Arc<_>` inside?

Reply:
nice advice, I will fix it.